node/src/client/mod.rs

```diff
@@ -181,6 +242,177 @@ impl<N: Network, C: ConsensusStorage<N>> Client<N, C> {
         }));
     }
 
+    /// Initializes solution verification.
+    fn initialize_solution_verification(&self) {
+        // Start the solution verification loop.
+        let node = self.clone();
+        self.handles.lock().push(tokio::spawn(async move {
+            loop {
+                // If the Ctrl-C handler registered the signal, stop the node.
+                if node.shutdown.load(Acquire) {
```

Comment:
I know this wasn't done in this PR, but I feel like a [CancellationToken](https://docs.rs/tokio-util/latest/tokio_util/sync/struct.CancellationToken.html) could be considered for every use case I've seen this `Arc<AtomicBool>` so far.